--- plugins/AccessRankingGA/lib/AccessRankingGA/Plugin.pm	2011-06-19 08:45:26.841301164 +0900
+++ ../../AccessRankingGA/lib/AccessRankingGA/Plugin.pm	2012-08-28 23:37:09.144219532 +0900
@@ -16,8 +16,8 @@
     use MT::Log;
     my $log = MT::Log->new;
     $log->message($msg);
-    $log->level( MT::Log::DEBUG() );
-    $log->class($class) if $class;
+	$log->level(MT::Log::DEBUG());
+	$log->class($class) if $class;
     $log->save or die $log->errstr;
 }
 
@@ -47,6 +47,7 @@
     my $conf = {
         user => $plugin->get_config_value( 'GA_username', 'blog:' . $blog_id ),
         pass => $plugin->get_config_value( 'GA_password', 'blog:' . $blog_id ),
+        api_key => $plugin->get_config_value( 'GA_api_key', 'blog:' . $blog_id ),
         profileid =>
           $plugin->get_config_value( 'GA_profile_id', 'blog:' . $blog_id ),
         maxresult =>
@@ -62,19 +63,17 @@
 
     my $token  = &get_token($conf);
     my $data   = &get_data( $token, $conf );
-    my $parser = XML::Simple->new( Forcearray => 1 );
+    my $parser = XML::Simple->new(ForceArray => 1);
     my $xml    = $parser->XMLin($data);
-    if ( MT->VERSION >= 5 ) {
-        foreach my $item ( @{ $xml->{entry} } ) {
-            $item->{title}[0] = decode( 'utf-8', $item->{title}[0] );
-            $item->{'dxp:dimension'}->{'ga:pageTitle'}->{value} =
-              decode( 'utf-8',
-                $item->{'dxp:dimension'}->{'ga:pageTitle'}->{value} );
-        }
-    }
+	if( MT->VERSION > 4) {
+		foreach my $entry ( @{ $xml->{entry} } ) {
+			$entry->{title}[0]->{content} = decode('utf-8', $entry->{title}[0]->{content});
+			$entry->{'dxp:dimension'}->{'ga:pageTitle'}->{value} = decode('utf-8', $entry->{'dxp:dimension'}->{'ga:pageTitle'}->{value});
+		}
+	}
 
-    my $json = to_json( $xml->{entry} );
-    return $json;
+	my $json = to_json( $xml->{entry} );
+	return $json;
 }
 
 sub get_token {
@@ -117,25 +116,26 @@
     my $ua =
       MT->new_ua( { agent => join( "/", $plugin->name, $plugin->version ) } );
     my $url =
-        "https://www.google.com/analytics/feeds/data?"
-      . "ids=ga%3A"
+        "https://www.googleapis.com/analytics/v3/data/ga?"
+      . "ids=ga:"
       . $conf->{profileid} . "&"
-      . "dimensions=ga%3ApagePath%2Cga%3ApageTitle&"
-      . "metrics=ga%3AuniquePageviews&"
-      . "sort=-ga%3AuniquePageviews&"
+      . "dimensions=ga:pagePath,ga:pageTitle&"
+      . "metrics=ga:uniquePageviews&"
+      . "sort=-ga:uniquePageviews&"
       . "start-date="
       . $conf->{start} . "&"
       . "end-date="
       . $conf->{end} . "&"
       . "max-results="
-      . $conf->{maxresult};
+      . $conf->{maxresult}. "&"
+			. $conf->{api_key};	
     if ( $conf->{ex_top} || $conf->{ex_path} && $conf->{ex_path} ne '' ) {
         $url .= "&filters=";
-        $url .= "ga%3ApagePath!%3D%2F;ga%3ApagePath!%3D%2Findex.html;"
+        $url .= "ga:pagePath!%3D%2F;ga:pagePath!%3D%2Findex.html;"
           if ( defined( $conf->{ex_top} ) );
         my @ex_paths = ( $conf->{ex_path} =~ /(\S+)/g );
         foreach my $exc (@ex_paths) {
-            $url .= "ga%3ApagePath!~" . $exc . ";";
+            $url .= "ga:pagePath!~" . $exc . ";";
         }
     }
 
